### ISSUE TITLE
Failed to install ruby 1.9.3 on OS X 10.10.3

### ISSUE THREAD

**Issue by jeremymodell** [sstephenson/rbenv#712]
EDIT: Weird, reinstalling xcode command line tools seems to have fixed the problem
Anyways, the previous problem is kept below

Trying to install ruby 1.9.3 on OS X 10.10.3 but it's not cooperating. Of note: had an earlier error from homebrew  that led me to grab  a legacy version of gcc called applegcc-42. The previous error goes as follows 

Anyways 
``` 
ERROR: This package must be compiled with GCC, but ruby-build couldn't
find a suitable `gcc` executable on your system. Please install GCC
and try again.

DETAILS: Apple no longer includes the official GCC compiler with Xcode
as of version 4.2. Instead, the `gcc` executable is a symlink to
`llvm-gcc`, a modified version of GCC which outputs LLVM bytecode.

For most programs the `llvm-gcc` compiler works fine. However,
versions of Ruby older than 1.9.3-p125 are incompatible with
`llvm-gcc`. To build older versions of Ruby you must have the official
GCC compiler installed on your system.

TO FIX THE PROBLEM: Install Homebrew's apple-gcc42 package with this
command: brew tap homebrew/dupes ; brew install apple-gcc42

You will need to install the official GCC compiler to build older
versions of Ruby even if you have installed Apple's Command Line Tools
for Xcode package. The Command Line Tools for Xcode package only
includes `llvm-gcc`.

BUILD FAILED (OS X 10.10.3 using ruby-build 20150319zf)
```
Then after installing applegcc-42 through homebrew the next error comes up and won't go away. 

```
 BUILD FAILED (OS X 10.10.3 using ruby-build 20150319zf)

Inspect or clean up the working tree at /var/folders/v0/n_ycymws751_0t2d0qshy54m0000gn/T/ruby-build.20150326020732.5520
```
The log goes as follows 
```
/var/folders/v0/n_ycymws751_0t2d0qshy54m0000gn/T/ruby-build.20150326020732.55202 ~
HTTP/1.1 200 OK
Content-Type: binary/octet-stream
Content-Length: 12223217
Connection: keep-alive
Date: Wed, 25 Mar 2015 17:47:50 GMT
Last-Modified: Fri, 23 May 2014 08:15:12 GMT
ETag: "8e2fef56185cfbaf29d0c8329fc77c05"
Accept-Ranges: bytes
Server: AmazonS3
Age: 44384
X-Cache: Hit from cloudfront
Via: 1.1 fd1988d68aa7c07df244e027c2b85034.cloudfront.net (CloudFront)
X-Amz-Cf-Id: aNO7cg3TQ1QfwGNOgoyTxotn_G7y_b4CB1E5Au7Rpkm66L9bHLY1Wg==

/var/folders/v0/n_ycymws751_0t2d0qshy54m0000gn/T/ruby-build.20150326020732.55202/ruby-1.9.3-p0 /var/folders/v0/n_ycymws751_0t2d0qshy54m0000gn/T/ruby-build.20150326020732.55202 ~
configure: WARNING: unrecognized options: --with-libyaml-dir
checking build system type... x86_64-apple-darwin14.3.0
checking host system type... x86_64-apple-darwin14.3.0
checking target system type... x86_64-apple-darwin14.3.0
checking whether the C compiler works... no
configure: error: in `/var/folders/v0/n_ycymws751_0t2d0qshy54m0000gn/T/ruby-build.20150326020732.55202/ruby-1.9.3-p0':
configure: error: C compiler cannot create executables
See `config.log' for more details
make: *** No targets specified and no makefile found.  Stop. 
```

**Comment by jeremymodell** [comment]
As per the compiler config log, also attached below 
```
Jeremy-Matos-Modells-MacBook-Pro: ~$cat /var/folders/v0/n_ycymws751_0t2d0qshy54m0000gn/T/ruby-build.20150326020732.55202/ruby-1.9.3-p0/config.log
This file contains any messages produced by compilers while
running configure, to aid debugging if configure makes a mistake.

It was created by configure, which was
generated by GNU Autoconf 2.68.  Invocation command line was

  $ ./configure --prefix=/Users/jeremymatos-modell/.rbenv/versions/1.9.3-p0 --with-libyaml-dir=/usr/local/opt/libyaml

## --------- ##
## Platform. ##
## --------- ##

hostname = Jeremy-Matos-Modells-MacBook-Pro.local
uname -m = x86_64
uname -r = 14.3.0
uname -s = Darwin
uname -v = Darwin Kernel Version 14.3.0: Tue Mar 10 20:10:46 PDT 2015; root:xnu-2782.20.44.1.1~1/RELEASE_X86_64

/usr/bin/uname -p = i386
/bin/uname -X     = unknown

/bin/arch              = unknown
/usr/bin/arch -k       = unknown
/usr/convex/getsysinfo = unknown
/usr/bin/hostinfo      = Mach kernel version:
	 Darwin Kernel Version 14.3.0: Tue Mar 10 20:10:46 PDT 2015; root:xnu-2782.20.44.1.1~1/RELEASE_X86_64
Kernel configured for up to 4 processors.
2 processors are physically available.
4 processors are logically available.
Processor type: i486 (Intel 80486)
Processors active: 0 1 2 3
Primary memory available: 8.00 gigabytes
Default processor set: 258 tasks, 1215 threads, 4 processors
Load average: 4.37, Mach factor: 0.92
/bin/machine           = unknown
/usr/bin/oslevel       = unknown
/bin/universe          = unknown

PATH: /usr/local/Cellar/rbenv/0.4.0/libexec
PATH: /Library/Frameworks/Python.framework/Versions/3.4/bin
PATH: /usr/local/bin
PATH: /usr/bin
PATH: /bin
PATH: /usr/sbin
PATH: /sbin
PATH: /opt/X11/bin
PATH: /usr/local/git/bin
PATH: /usr/texbin


## ----------- ##
## Core tests. ##
## ----------- ##

configure:2764: checking build system type
configure:2778: result: x86_64-apple-darwin14.3.0
configure:2849: checking host system type
configure:2862: result: x86_64-apple-darwin14.3.0
configure:2882: checking target system type
configure:2895: result: x86_64-apple-darwin14.3.0
configure:3376: checking for C compiler version
configure:3385: /usr/local/bin/gcc-4.2 --version >&5
i686-apple-darwin11-gcc-4.2.1 (GCC) 4.2.1 (Apple Inc. build 5666) (dot 3)
Copyright (C) 2007 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

configure:3396: $? = 0
configure:3385: /usr/local/bin/gcc-4.2 -v >&5
Using built-in specs.
Target: i686-apple-darwin11
Configured with: /Volumes/Media/Builds/gcc-5666.3/build/obj/src/configure --disable-checking --prefix=/usr --mandir=/share/man --enable-languages=c,objc,c++,obj-c++,fortran --program-transform-name=/^[cg][^.-]*$/s/$/-4.2/ --with-slibdir=/usr/lib --build=i686-apple-darwin11 --program-prefix=i686-apple-darwin11- --host=x86_64-apple-darwin11 --target=i686-apple-darwin11 --with-gxx-include-dir=/include/c++/4.2.1
Thread model: posix
gcc version 4.2.1 (Apple Inc. build 5666) (dot 3)
configure:3396: $? = 0
configure:3385: /usr/local/bin/gcc-4.2 -V >&5
gcc-4.2: argument to `-V' is missing
configure:3396: $? = 1
configure:3385: /usr/local/bin/gcc-4.2 -qversion >&5
i686-apple-darwin11-gcc-4.2.1: no input files
configure:3396: $? = 1
configure:3416: checking whether the C compiler works
configure:3438: /usr/local/bin/gcc-4.2  -O3 -Wno-error=shorten-64-to-32  -I/Users/jeremymatos-modell/.rbenv/versions/1.9.3-p0/include  -L/Users/jeremymatos-modell/.rbenv/versions/1.9.3-p0/lib  conftest.c  >&5
ld: warning: directory not found for option '-L/Users/jeremymatos-modell/.rbenv/versions/1.9.3-p0/lib'
ld: library not found for -lcrt1.10.6.o
collect2: ld returned 1 exit status
configure:3442: $? = 1
configure:3480: result: no
configure: failed program was:
| /* confdefs.h */
| #define PACKAGE_NAME ""
| #define PACKAGE_TARNAME ""
| #define PACKAGE_VERSION ""
| #define PACKAGE_STRING ""
| #define PACKAGE_BUGREPORT ""
| #define PACKAGE_URL ""
| #define CANONICALIZATION_FOR_MATHN 1
| /* end confdefs.h.  */
| 
| int
| main ()
| {
| 
|   ;
|   return 0;
| }
configure:3485: error: in `/var/folders/v0/n_ycymws751_0t2d0qshy54m0000gn/T/ruby-build.20150326020732.55202/ruby-1.9.3-p0':
configure:3487: error: C compiler cannot create executables
See `config.log' for more details

## ---------------- ##
## Cache variables. ##
## ---------------- ##

ac_cv_build=x86_64-apple-darwin14.3.0
ac_cv_env_CCC_set=
ac_cv_env_CCC_value=
ac_cv_env_CC_set=set
ac_cv_env_CC_value=/usr/local/bin/gcc-4.2
ac_cv_env_CFLAGS_set=set
ac_cv_env_CFLAGS_value=' -O3 -Wno-error=shorten-64-to-32 '
ac_cv_env_CPPFLAGS_set=set
ac_cv_env_CPPFLAGS_value='-I/Users/jeremymatos-modell/.rbenv/versions/1.9.3-p0/include '
ac_cv_env_CPP_set=
ac_cv_env_CPP_value=
ac_cv_env_CXXFLAGS_set=
ac_cv_env_CXXFLAGS_value=
ac_cv_env_CXX_set=
ac_cv_env_CXX_value=
ac_cv_env_LDFLAGS_set=set
ac_cv_env_LDFLAGS_value='-L/Users/jeremymatos-modell/.rbenv/versions/1.9.3-p0/lib '
ac_cv_env_LIBS_set=
ac_cv_env_LIBS_value=
ac_cv_env_build_alias_set=
ac_cv_env_build_alias_value=
ac_cv_env_host_alias_set=
ac_cv_env_host_alias_value=
ac_cv_env_target_alias_set=
ac_cv_env_target_alias_value=
ac_cv_host=x86_64-apple-darwin14.3.0
ac_cv_prog_CC=/usr/local/bin/gcc-4.2
ac_cv_target=x86_64-apple-darwin14.3.0

## ----------------- ##
## Output variables. ##
## ----------------- ##

ALLOCA=''
AR=''
ARCHFILE=''
ARCH_FLAG=''
AS=''
ASFLAGS=''
BASERUBY='ruby'
BUILTIN_ENCOBJS=''
BUILTIN_TRANSOBJS=''
BUILTIN_TRANSSRCS=''
CAPITARGET=''
CC='/usr/local/bin/gcc-4.2'
CCDLFLAGS=''
CFLAGS=' -O3 -Wno-error=shorten-64-to-32 '
CHDIR=''
COMMON_HEADERS=''
COMMON_LIBS=''
COMMON_MACROS=''
COUTFLAG=''
CP=''
CPP=''
CPPFLAGS='-I/Users/jeremymatos-modell/.rbenv/versions/1.9.3-p0/include '
CPPOUTFILE=''
CXX=''
CXXFLAGS=''
DEFS=''
DLDFLAGS=''
DLDLIBS=''
DLEXT2=''
DLEXT=''
DLLWRAP=''
DOT=''
DOXYGEN=''
ECHO_C='\c'
ECHO_N=''
ECHO_T=''
EGREP=''
ENABLE_SHARED=''
EXECUTABLE_EXTS=''
EXEEXT=''
EXPORT_PREFIX=''
EXTOUT=''
EXTSTATIC=''
GCC=''
GNU_LD=''
GREP=''
INSTALLDOC=''
INSTALL_DATA=''
INSTALL_PROGRAM=''
INSTALL_SCRIPT=''
LDFLAGS='-L/Users/jeremymatos-modell/.rbenv/versions/1.9.3-p0/lib '
LDSHARED=''
LDSHAREDXX=''
LIBEXT=''
LIBOBJS=''
LIBPATHENV=''
LIBPATHFLAG=''
LIBRUBY=''
LIBRUBYARG=''
LIBRUBYARG_SHARED=''
LIBRUBYARG_STATIC=''
LIBRUBY_A=''
LIBRUBY_ALIASES=''
LIBRUBY_DLDFLAGS=''
LIBRUBY_LDSHARED=''
LIBRUBY_RELATIVE=''
LIBRUBY_SO=''
LIBS=''
LINK_SO=''
LN_S=''
LTLIBOBJS=''
MAINLIBS=''
MAJOR='1'
MAKEDIRS=''
MAKEFILES=''
MANTYPE=''
MINIOBJS=''
MINIRUBY=''
MINOR='9'
MKDIR_P=''
NM=''
NROFF=''
NULLCMD=''
OBJCOPY=''
OBJDUMP=''
OBJEXT=''
OUTFLAG=''
PACKAGE=''
PACKAGE_BUGREPORT=''
PACKAGE_NAME=''
PACKAGE_STRING=''
PACKAGE_TARNAME=''
PACKAGE_URL=''
PACKAGE_VERSION=''
PATH_SEPARATOR=':'
PKG_CONFIG=''
PREP=''
RANLIB=''
RDOCTARGET=''
RI_BASE_NAME=''
RM=''
RMALL=''
RMDIR=''
RMDIRS=''
RPATHFLAG=''
RUBYW_BASE_NAME='rubyw'
RUBYW_INSTALL_NAME=''
RUBY_BASE_NAME='ruby'
RUBY_INSTALL_NAME=''
RUBY_PROGRAM_VERSION='1.9.3'
RUBY_RELEASE_DATE='2011-10-30'
RUBY_SO_NAME=''
RUNRUBY=''
SET_MAKE=''
SHELL='/bin/sh'
SOLIBS=''
STATIC=''
STRIP=''
SYMBOL_PREFIX=''
TEENY='1'
TEST_RUNNABLE=''
THREAD_MODEL=''
TRY_LINK=''
UNIVERSAL_ARCHNAMES=''
UNIVERSAL_INTS=''
USE_RUBYGEMS=''
WERRORFLAG=''
WINDRES=''
XCFLAGS=''
XLDFLAGS=''
XRUBY=''
XRUBY_LIBDIR=''
XRUBY_RUBYHDRDIR=''
XRUBY_RUBYLIBDIR=''
ac_ct_CC=''
ac_ct_CXX=''
ac_ct_OBJCOPY=''
ac_ct_OBJDUMP=''
arch=''
bindir='${exec_prefix}/bin'
build='x86_64-apple-darwin14.3.0'
build_alias=''
build_cpu='x86_64'
build_os='darwin14.3.0'
build_vendor='apple'
cflags=''
configure_args=''
cppflags=''
cxxflags=' ${optflags} ${debugflags} ${warnflags}'
datadir='${datarootdir}'
datarootdir='${prefix}/share'
debugflags=''
docdir='${datarootdir}/doc/${PACKAGE}'
dvidir='${docdir}'
exec=''
exec_prefix='NONE'
host='x86_64-apple-darwin14.3.0'
host_alias=''
host_cpu='x86_64'
host_os='darwin14.3.0'
host_vendor='apple'
htmldir='${docdir}'
includedir='${prefix}/include'
infodir='${datarootdir}/info'
libdir='${exec_prefix}/lib'
libexecdir='${exec_prefix}/libexec'
localedir='${datarootdir}/locale'
localstatedir='${prefix}/var'
mandir='${datarootdir}/man'
oldincludedir='/usr/include'
optflags=''
pdfdir='${docdir}'
prefix='/Users/jeremymatos-modell/.rbenv/versions/1.9.3-p0'
program_transform_name='s&^&&'
psdir='${docdir}'
ridir=''
ruby_pc=''
ruby_version=''
rubyhdrdir=''
rubylibprefix=''
rubyw_install_name=''
sbindir='${exec_prefix}/sbin'
setup=''
sharedstatedir='${prefix}/com'
sitearch=''
sitedir=''
sitehdrdir=''
sysconfdir='${prefix}/etc'
target='x86_64-apple-darwin14.3.0'
target_alias=''
target_cpu='x86_64'
target_os='darwin14.3.0'
target_vendor='apple'
try_header=''
vendordir=''
vendorhdrdir=''
warnflags=''

## ----------- ##
## confdefs.h. ##
## ----------- ##

/* confdefs.h */
#define PACKAGE_NAME ""
#define PACKAGE_TARNAME ""
#define PACKAGE_VERSION ""
#define PACKAGE_STRING ""
#define PACKAGE_BUGREPORT ""
#define PACKAGE_URL ""
#define CANONICALIZATION_FOR_MATHN 1

configure: exit 77
```

**Comment by hsbt** [comment]
Ruby 1.9.3 is EOL now. ruby core team no longer support Ruby 1.9.3. There is no solution of this issue.
